data_structures/linked_list/circular_doubly_linked_list.c

```diff
@@ -0,0 +1,229 @@
+/*
+ * circular_doubly_linked_list.c
+ *
+ * @details
+ Circular Doubly Linked List combines the properties of a doubly linked list and a circular linked list in which two consecutive elements are linked or connected by the previous and next pointer and the last node points to the first node via the next pointer and the first node points last node via the previous pointer.
+
+ In this implementation, the menu driven code is written along with functions to insert at start, insrt at last, delete first node, delete last node and display list functions are coded.
+
+ * Author: [Sahil Kandhare](https://github.com/SahilK-027)
+ * Created on 12/10/2002
+ *
+ *
+ * Cicular doubly linked list in C
+ *
+ * Compiled in Visual Studio Code
+ *
+ */
+
+#include <stdio.h>   // for IO operations
+#include <stdlib.h>  // for including functions involving memory allocation( eg. malloc,calloc)
+/**
+ * @brief Node Structure Cicular Doubly linked list.
+ */
+struct node
+{
+    struct node *prev, *next; // List pointers
+    int value;                // Data stored on each node
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Requires adding the `inttypes.h` library/header.** **Let us know if you need any help here or on our [Discord server](https://the-algorithms.com/discord).** 🙂